### ISSUE TITLE
Data loss when parsing numbers

### ISSUE THREAD

**Issue by xb** [flori/json#273]
## Expected

``` ruby
2.2.4 :001 > require 'bigdecimal'; require 'json'
 => true 
2.2.4 :002 > JSON.parse('[0.001e500]')[0].to_s
 => "0.1E498" 
2.2.4 :003 > JSON.parse('[0.11111111111111111111111222222222222222222222222]')[0].to_s
 => "0.11111111111111111111111222222222222222222222222E0" 
```
## Observed

``` ruby
2.2.4 :001 > require 'bigdecimal'; require 'json'
 => true 
2.2.4 :002 > JSON.parse('[0.001e500]')[0].to_s
 => "Infinity"            # The number is actually not infinity
2.2.4 :003 > JSON.parse('[0.11111111111111111111111222222222222222222222222]')[0].to_s
 => "0.1111111111111111"  # The '222222' section is missing
```

**Comment by Umofomia** [comment]
It looks like this issue (and an older issue #219 that reported the same thing) may have been addressed by 7e240ed5765867bfbdd166435f941a4cebed8cd8, which appears to be adapted from #223 by @jaschob.

@flori When do you think v2.1.0 will be published so that we can make use of the new `decimal_class` parser? This capability is extremely important for dealing with any APIs that expose monetary values.

**Comment by Umofomia** [comment]
Just noting that it looks like this got released in v2.0.3 (44b392b5ede944f73cfbdf65afb9d4dd972e23f5). @xb did note in #320 that there may be an issue that causes a segmentation fault though.

**Comment by flori** [comment]
The decimal_class option should be working now in the newly released version 2.1.0.

**Comment by Umofomia** [comment]
Thanks @flori!
